app/src/main/java/com/duckduckgo/app/onboarding/di/WelcomePageModule.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2018 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.onboarding.di
+
+import android.content.Context
+import com.duckduckgo.app.global.DefaultRoleBrowserDialogExperiment
+import com.duckduckgo.app.global.RealDefaultRoleBrowserDialogExperiment
+import com.duckduckgo.app.global.install.AppInstallStore
+import com.duckduckgo.app.onboarding.ui.page.WelcomePageViewModelFactory
+import com.duckduckgo.app.statistics.VariantManager
+import com.duckduckgo.app.statistics.pixels.Pixel
+import dagger.Module
+import dagger.Provides
+
+@Module
+class WelcomePageModule {
```

Comment:
One question: I see this new approach and I want to know more. would you suggest to create `@Module` per Activity/Fragment? why that instead of per feature? Asking that because we have `OnboardingModule` and I think this could be maybe there?

Reply:
The setup we currently have has every DI instance in one single component (scope) which is the app scope. However, we are creating one component (scope) per activity/fragment that are completely empty.
This means that everything in DI is visible to every component, even dependencies that are very specific to say a activity/fragment, as they are placed in the app scope. which is kinda anti-pattern for DI.

I thought to take baby steps in this PR to a better DI organization. In this, `WelcomePageModule` has only the dependencies for the welcome Page (eg. `welcomePageViewModelFactory`. Nobody else should see those dependencies in DI or be able to get them. And `WelcomePageModule` is added to the `WelcomePage` scope.

Hope this clarifies